File: eecs168/2020fall/MWF/2020.11.09.classes-part-3/Account/Account.cpp
//Account.cpp
#include "Account.h"

Account::Account()
{
	//initialize all member variables
	balance = 0;
}

Account::Account(double startingBalance)
{
	if(startingBalance > 0)
	{
		balance = startingBalance;
	}
	else
	{
		balance = 0;
	}
}

bool Account::deposit(double amount)
{
	if(amount > 0)
	{	
		balance = balance + amount;
		return (true);
	}	
	else
	{
		return (false);
	}
}

bool Account::withdraw(double amount)
{
	if(amount <= balance && amount > 0)
	{
		balance = balance - amount;
		return (true);
	}
	else
	{
		return(false);
	}
}

double Account::checkBalance()
{
	return(balance);
}

File: eecs168/2025spring/TR/2025.04.22.classes-part-2/account/account.py
#account.py

class Account:
    def __init__(self):
        self._balance = 0
        self.test_variable = 0

    def deposit(self, amount):
        if amount > 0:
            self._balance += amount
        else:
            pass

    def check_balance(self):
        return self._balance
